lib/OpenQA/Worker/Jobs.pm

```diff
@@ -825,14 +829,21 @@ sub read_result_file($$) {
 }
 
 sub backend_running {
-    return $worker;
+    return $worker->{child}->is_running;
 }
 
 sub check_backend {
     log_debug("checking backend state");
-    my $res = engine_check;
-    if ($res && $res ne 'ok') {
-        stop_job($res);
+    unless ($worker->{child}->is_running()) {
+        if ($worker->{child}->errored) {
+            stop_job('died');
+        }
+        else {
+            stop_job('done');
+        }
+    }
+    else {
+        log_debug("backend is running!");
     }
```

Comment:
How about an early return to reduce indentation: `return log_debug('backend is running') if $worker->{child}->is_running();`
Especially because double negation is not easy to read: `unless ... else` :-/

Reply:
done  :-)

👍